commit 2d4ef556d34ddc60a62c339b9e20e9f1594d80f6
Author: Adlertz, Niclas <niclas.adlertz@intel.com>
Date:   Wed Nov 19 14:33:14 2025 +0100

    Do not depend on Zarith

diff --git a/scripts/mk_util.py b/scripts/mk_util.py
index c1070e62a..7f2055772 100644
--- a/scripts/mk_util.py
+++ b/scripts/mk_util.py
@@ -1998,8 +1998,8 @@ class MLComponent(Component):
                 OCAML_FLAGS += '-g'
 
             if OCAMLFIND:
-                OCAMLCF = OCAMLFIND + ' ' + 'ocamlc -package zarith' + ' ' + OCAML_FLAGS
-                OCAMLOPTF = OCAMLFIND + ' ' + 'ocamlopt -package zarith' + ' ' + OCAML_FLAGS
+                OCAMLCF = OCAMLFIND + ' ' + 'ocamlc' + ' ' + OCAML_FLAGS
+                OCAMLOPTF = OCAMLFIND + ' ' + 'ocamlopt' + ' ' + OCAML_FLAGS
             else:
                 OCAMLCF = OCAMLC + ' ' + OCAML_FLAGS
                 OCAMLOPTF = OCAMLOPT + ' ' + OCAML_FLAGS
@@ -2339,7 +2339,7 @@ class MLExampleComponent(ExampleComponent):
             out.write('\tocamlfind %s ' % OCAMLC)
             if DEBUG_MODE:
                 out.write('-g ')
-            out.write('-custom -o ml_example.byte -package zarith -I api/ml -cclib "-L. -lpthread -lstdc++ -lz3" -linkpkg z3ml.cma')
+            out.write('-custom -o ml_example.byte  -I api/ml -cclib "-L. -lpthread -lstdc++ -lz3" -linkpkg z3ml.cma')
             for mlfile in get_ml_files(self.ex_dir):
                 out.write(' %s/%s' % (self.to_ex_dir, mlfile))
             out.write('\n')
@@ -2350,7 +2350,7 @@ class MLExampleComponent(ExampleComponent):
             out.write('\tocamlfind %s ' % OCAMLOPT)
             if DEBUG_MODE:
                 out.write('-g ')
-            out.write('-o ml_example$(EXE_EXT) -package zarith -I api/ml -cclib "-L. -lpthread -lstdc++ -lz3" -linkpkg z3ml.cmxa')
+            out.write('-o ml_example$(EXE_EXT) -I api/ml -cclib "-L. -lpthread -lstdc++ -lz3" -linkpkg z3ml.cmxa')
             for mlfile in get_ml_files(self.ex_dir):
                 out.write(' %s/%s' % (self.to_ex_dir, mlfile))
             out.write('\n')
diff --git a/src/api/ml/CMakeLists.txt b/src/api/ml/CMakeLists.txt
index e3f3168ad..55626cbc3 100644
--- a/src/api/ml/CMakeLists.txt
+++ b/src/api/ml/CMakeLists.txt
@@ -57,7 +57,7 @@ add_custom_command(
 )
 
 
-set(z3ml_common_flags  "-package" "zarith"
+set(z3ml_common_flags
                        "-I" "${z3ml_bin}")
 
                        # add_custom_command(
@@ -185,7 +185,6 @@ endif()
 
 set(ocamlmklib_flags "-o" "z3ml"
                      "-ocamlcflags" "-bin-annot"
-                     "-package" "zarith"
                      ${c_lib_deps}
                      "-dllpath" "${libz3_path}"
                      "-L${ocaml_stublibs_path}"
@@ -278,7 +277,6 @@ add_custom_command(
   TARGET build_z3_ocaml_bindings POST_BUILD
   COMMAND "${OCAMLFIND}" ocamlc
     -o "${z3ml_bin}/ml_example.byte"
-    -package zarith
     -linkpkg
     -I "${z3ml_bin}"
     -dllpath "${z3ml_bin}"
@@ -293,7 +291,6 @@ add_custom_command(
   TARGET build_z3_ocaml_bindings POST_BUILD
   COMMAND "${OCAMLFIND}" ocamlopt
     -o "${z3ml_bin}/ml_example"
-    -package zarith
     -linkpkg
     -I "${z3ml_bin}"
     "${z3ml_bin}/z3ml.cmxa"
@@ -327,4 +324,4 @@ add_custom_command(
 #     COMMAND mv "${z3ml_bin}/dllz3ml.pre.so" "${z3ml_bin}/dllz3ml.so"
 #     DEPENDS "${z3ml_bin}/dllz3ml.so"
 #   )
-# endif()
\ No newline at end of file
+# endif()
diff --git a/src/api/ml/META.in b/src/api/ml/META.in
index ad27e6b0b..33ebbef20 100644
--- a/src/api/ml/META.in
+++ b/src/api/ml/META.in
@@ -1,7 +1,6 @@
 # META file for the "z3" package:
 version = "@VERSION@"
 description = "Z3 Theorem Prover (OCaml API)"
-requires = "zarith threads"
 archive(byte) = "z3ml.cma"
 archive(native) = "z3ml.cmxa"
 archive(byte,plugin) = "z3ml.cma"
diff --git a/src/api/ml/z3.ml b/src/api/ml/z3.ml
index 8f46e551c..c622571e9 100644
--- a/src/api/ml/z3.ml
+++ b/src/api/ml/z3.ml
@@ -1037,13 +1037,6 @@ struct
       | true, v -> v
       | false, _ -> raise (Error "Conversion failed.")
 
-    let get_big_int (x:expr) =
-      if is_int_numeral x then
-        let s = (Z3native.get_numeral_string (Expr.gc x) x) in
-        Z.of_string s
-      else
-        raise (Error "Conversion failed.")
-
     let numeral_to_string (x:expr) = Z3native.get_numeral_string (Expr.gc x) x
     let mk_const (ctx:context) (name:Symbol.symbol) = Expr.mk_const ctx name (mk_sort ctx)
     let mk_const_s (ctx:context) (name:string) = mk_const ctx (Symbol.mk_string ctx name)
@@ -1061,13 +1054,6 @@ struct
     let get_numerator x = Z3native.get_numerator (Expr.gc x) x
     let get_denominator x = Z3native.get_denominator (Expr.gc x) x
 
-    let get_ratio x =
-      if is_rat_numeral x then
-        let s = Z3native.get_numeral_string (Expr.gc x) x in
-        Q.of_string s
-      else
-        raise (Error "Conversion failed.")
-
     let to_decimal_string (x:expr) (precision:int) = Z3native.get_numeral_decimal_string (Expr.gc x) x precision
     let numeral_to_string (x:expr) = Z3native.get_numeral_string (Expr.gc x) x
     let mk_const (ctx:context) (name:Symbol.symbol) = Expr.mk_const ctx name (mk_sort ctx)
diff --git a/src/api/ml/z3.mli b/src/api/ml/z3.mli
index b7a74049f..da0acf2db 100644
--- a/src/api/ml/z3.mli
+++ b/src/api/ml/z3.mli
@@ -1194,9 +1194,6 @@ sig
     (** Create a new integer sort. *)
     val mk_sort : context -> Sort.sort
 
-    (** Get a big_int from an integer numeral *)
-    val get_big_int : Expr.expr -> Z.t
-
     (** Returns a string representation of a numeral. *)
     val numeral_to_string : Expr.expr -> string
 
@@ -1252,9 +1249,6 @@ sig
     (** The denominator of a rational numeral. *)
     val get_denominator : Expr.expr -> Expr.expr
 
-    (** Get a ratio from a real numeral *)
-    val get_ratio : Expr.expr -> Q.t
-
     (** Returns a string representation in decimal notation.
         The result has at most as many decimal places as indicated by the int argument.*)
     val to_decimal_string : Expr.expr-> int -> string
